### ISSUE TITLE
Cookie confusion

### ISSUE THREAD

**Issue by tjsandberg** [krcowles/ktesa#29]
I've been trying to figure out how cookies: how they're supposed to work, how they actually work on commercial websites, and how they actually work on our site. The first question is a general one: Is it possible to be logged into a website if the user does not accept/allow cookies for the site?

**Comment by krcowles** [comment]
Good question! Yes, you can log in to a site without cookies. The only issue you would have is that when you re-visit the site, you would not be automatically logged in: that is also how it works on our site. Our site is a ‘minimalist’ regarding cookies. We only save the username. Many commercial sites want to retain other information about you and have a lot more cookie data.

**Comment by krcowles** [comment]
Good question. Yes, you can log in to a site without cookies. The only issue you would have is that when you re-visit the site, you would not be automatically logged back in: that is also how it works on our site. Our site is a ‘minimalist’ regarding cookies. We only save the username. Many commercial sites want to retain other information about you and have a lot more cookie data.

**Comment by tjsandberg** [comment]
I don't get it. What does it even mean to be logged in if I have rejected cookies? Consider the following example: I go to the site, reject cookies, log in, then click a link to go to another page on the site. With no cookies, how does the site know I'm logged in? How does the site even know it's me?

**Comment by krcowles** [comment]
We also use secure sessions for login. Once you are logged in, you can do whatever a user does, on any page. The session data tells the page you are logged in. When you leave (and session data expires), you can re-login and have full user access again. As long as you have logged in, there is no problem accessing pages. If you reject cookies and later, when session data expires, wish to revisit the site, you would have to login again instead of having that happen automagically.

**Comment by tjsandberg** [comment]
Bear with me as I try to comprehend the big picture as well as our implementation. As I understand it, a website should inform users about whether the site keeps any of their personal information, allow users to opt-out, etc. I believe our intent is to _not_ store any user information unless the user creates an account.

So maybe we can get away with doing the following:

1. Create a privacy policy that explains what we do regarding user data. Among other things, this could include:
   1. A description of how we use cookies, a database, etc., to retain user data and make the site work better.
   1. A statement that we store nothing about users who don't create an account.
1. Make the privacy policy available from the Help menu or whatever.
1. Warn users when they create an account that we are storing the information they provide.

**Comment by krcowles** [comment]
No problem. Your opening statement is absolutely correct. Up to now, since cookie data is only *potentially* kept when a user registers, that's when the cookie 'template' appears - on the registration (and renew) page. This is in accordance with how I've seen other sites do it. In fact, the template was created from an online example. In that cookie 'template', it explains *what* data is stored and allows the user to accept/reject (this satisfies your item 3). It explains what happens if the user chooses to reject. I didn't see the need to create a separate, explanatory privacy policy when no cookie data is being kept, but if you feel that's important, we can certainly add it. To see what the cookie 'template' looks like, go to the user registration page. That can also be modified if it is missing something you mentioned.

**Comment by krcowles** [comment]
Is this ok to close, or would you like to comment on the above? Note, if you want the latest registration page with the cosmetic and functional changes, visit the test site [https://nmhikes.com/master_6f0d8c4](url) and access the page via accounts/registration.php (or logout and begin the process of 'joining' as a new member)

**Comment by tjsandberg** [comment]
Sorry, I still think the site's behavior is not right, but I don't have the energy to defend my opinion on the topic. I guess that means it's up to you whether to close the issue.

**Comment by krcowles** [comment]
No problem. If you have, or come up with, an alternative approach, I'm happy to implement it. I'll give it a few weeks then I'll close the issue.

**Comment by krcowles** [comment]
I have created a Privacy Policy, now available under the 'Help' menu. It includes the elements you described above. I have also attached a link to the Privacy Policy on the registration page so that anyone can check it out when registering. NOTE: I discovered that no registration or login is possible when cookies are turned off. This is due to the fact that login credentials are held in secure php session variables. Those variables cannot be accessed from another page since the new page can't get the session id without cookies.
